### PR TITLE
net-irc/inspircd: Add v3.13.0

### PR DESCRIPTION
**Pull request by clinew**
Package-Manager: Portage-3.0.30, Repoman-3.0.3
Signed-off-by: Wade Cline <wadecline@hotmail.com>

Re-attempt of #25295:

 *   Add SystemD unit file
 *   Add logrotate file
 *   Move apparmor example to share directory
 *   Remove irrelevant start-up script
 *   Remove old 2.x -> 3.x upgrade warning

### CODE REVIEW

File: net-irc/inspircd/inspircd-3.13.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	newconfd "${FILESDIR}/${PN}.confd" "${PN}"
+	systemd_dounit .configure/inspircd.service
+
+	keepdir "/var/log/${PN}"
+	insinto "/etc/logrotate.d"
+	newins .configure/logrotate "${PN}"
+
+	diropts -o"${PN}" -g"${PN}" -m0700
+	keepdir "/var/lib/${PN}/data"
+}
```
Comment (ionenwks) [inline net-irc/inspircd/inspircd-3.13.0.ebuild:103]:
```
 * QA Notice: This ebuild installs into paths that should be created at runtime.
 *  To fix, simply do not install into these directories.  Instead, your package
 *  should create dirs on the fly at runtime as needed via init scripts/etc...
 * 
 *   run
 *   run/inspircd
```
`rmdir "${ED}"/run{/inspircd,} || die` would silence this

Haven't tried inits yet but guess runtime handling looks fine for both openrc and systemd (which does this differently)

File: net-irc/inspircd/inspircd-3.13.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit systemd toolchain-funcs
+
+DESCRIPTION="Inspire IRCd - The Stable, High-Performance Modular IRCd"
+HOMEPAGE="https://www.inspircd.org/"
+SRC_URI="https://github.com/inspircd/inspircd/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SRC_URI+=" https://github.com/clinew/gentoo-distfiles/raw/master/inspircd-${PV}-fix-path-builds.patch.bz2 https://github.com/clinew/gentoo-distfiles/raw/master/inspircd-${PV}-gentooize-logrotate.patch.bz2"
```
Comment (ionenwks) [inline net-irc/inspircd/inspircd-3.13.0.ebuild:11]: just style but this line sure is long, didn't even see the 2nd patch -- may want a 2nd += or do it like:
```bash
SRC_URI="
	https://github.com/inspircd/inspircd/archive/v${PV}.tar.gz -> ${P}.tar.gz
	https://github.com/clinew/gentoo-distfiles/raw/master/inspircd-${PV}-fix-path-builds.patch.bz2
	https://github.com/clinew/gentoo-distfiles/raw/master/inspircd-${PV}-gentooize-logrotate.patch.bz2"
```

File: net-irc/inspircd/inspircd-3.13.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+DOCS=( docs/. .configure/apparmor )
+PATCHES=(
+	"${WORKDIR}"/${P}-fix-path-builds.patch
+	"${WORKDIR}"/${P}-gentooize-logrotate.patch)
+
+src_prepare() {
+	default
+
+	# Patch the inspircd launcher with the inspircd user
+	sed -i -e "s/@UID@/${PN}/" "make/template/${PN}" || die
```
Comment (ionenwks) [inline net-irc/inspircd/inspircd-3.13.0.ebuild:44]: This sed isn't doing anything given the file doesn't contain `@UID@`, it wasn't either on previous version. Should need to be either updated or removed.

File: net-irc/inspircd/inspircd-3.13.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+		--binary-dir="/usr/bin"
+		--data-dir="/var/lib/${PN}/data"
+		--example-dir="/usr/share/doc/${PV}"
+		--manual-dir="/usr/share/man"
+		--module-dir="/usr/$(get_libdir)/${PN}/modules")
+	CXX="$(tc-getCXX)" ./configure "${myconf[@]}"
+}
+
+src_compile() {
+	emake LDFLAGS="${LDFLAGS}" CXXFLAGS="${CXXFLAGS}" $(usex debug 'INSPIRCD_DEBUG=2' '') INSPIRCD_VERBOSE=1
```
Comment (ionenwks) [inline net-irc/inspircd/inspircd-3.13.0.ebuild:84]: In EAPI=8, `$(usex)` with empty '' can be replaced use**v**, aka `$(usev debug INSPIRCD_DEBUG=2)`

**Review by ionenwks — COMMENTED** [review]
Still works fine with my server's configuration :+1:, but from a quick look there's a few things

File: net-irc/inspircd/inspircd-3.13.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+	# The first configuration run enables certain "extra" InspIRCd
+	# modules, the second run generates the actual makefile.
+	if [[ -n "${extras}" ]]; then
+		./configure --enable-extras=${extras%,}
+	fi
+
+	local myconf=(
+		--disable-auto-extras
+		--uid ${PN} --gid ${PN}
```
[the remark this answers is not in the document]
Comment (clinew) [reply net-irc/inspircd/inspircd-3.13.0.ebuild:73]: I put `--uid` and `--gid` there to set the username configuration in some of the scripts (`logrotate`, for one).  I didn't see that the installed files' ownership had changed; not sure why I didn't see Portage nagging.  I think I'll try disabling these flags and just add in username configuration to the patch file.

File: net-irc/inspircd/inspircd-3.13.0.ebuild (hunk 1 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-3.13.0.ebuild:103]: Should be easy enough.

File: net-irc/inspircd/inspircd-3.13.0.ebuild (hunk 3 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-3.13.0.ebuild:44]: Possibly a holdover from 2.x.  My understanding is that best practices are to put changes like this in a patch file anyways.  Will remove it.

File: net-irc/inspircd/inspircd-3.13.0.ebuild (hunk 4 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-3.13.0.ebuild:84]: Sounds good, will replace it.

File: net-irc/inspircd/inspircd-3.13.0.ebuild (hunk 2 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-3.13.0.ebuild:11]: Agreed; I'll fix it up.

**Comment by clinew** [comment]
I refactored the second patch to fix the template files in general.  This should be fixed, and all other items addressed (will leave it to you to confirm).

File: net-irc/inspircd/inspircd-3.13.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+DEPEND="${RDEPEND}"
+
+DOCS=( docs/. .configure/apparmor )
+PATCHES=(
+	"${WORKDIR}"/${P}-fix-path-builds.patch
+	"${WORKDIR}"/${P}-fix-template-files.patch)
+
+src_prepare() {
+	default
+}
```
Comment (ionenwks) [inline net-irc/inspircd/inspircd-3.13.0.ebuild:44]: I'll just drop this on merge, calling default is what is done by default :eyes:

**Review by ionenwks — APPROVED** [review]
Thanks.

File: net-irc/inspircd/inspircd-3.13.0.ebuild (hunk 6 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-3.13.0.ebuild:44]: D'oh :donut:!  ...No donut emoji?!  Boo!

Anyways, thanks for the review and merge!

**Pull request closed without merging**